Microsoft.Toolkit.Mvvm/Messaging/StrongReferenceMessenger.cs

```diff
@@ -358,10 +355,7 @@ public void UnregisterAll<TToken>(object recipient, TToken token)
                     return message;
                 }
 
-                handlers = ArrayPool<object>.Shared.Rent(totalHandlersCount);
-                recipients = ArrayPool<object>.Shared.Rent(totalHandlersCount);
-                handlersRef = ref handlers[0];
-                recipientsRef = ref recipients[0];
+                pairs = ArrayPool<Object2>.Shared.Rent(totalHandlersCount);
```

Comment:
You may get better pooling behavior in real world apps by using `ArrayPool<Object>` instead of `ArrayPool<Object2>`.

`ArrayPool<Object2>` is guaranteed to be used just for this library. `ArrayPool<object>` has a chance of being reused by other parts of the app.

Reply:
Thanks for taking a look! I did refactor all the other `ArrayPool<T>` usages to just use `object`, for the reason you mentioned, my rationale in this specific case was mostly that I'd expect this method to be called quite often in applications, and most importantly that if I used `object` here I'd basically need to either just box every pair, or just make the `Object2` type a class, which would still mean I'd end up allocating all those objects every single time (whereas the whole `Send` method right now has 0 amortized allocation) 🤔

I was thinking especially the fact that this method would be called often would justify these pooled arrays only being used here? What do you think?